frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card-body.tsx

```diff
@@ -43,15 +43,38 @@ export const TopConsumersBody: React.FC<TopConsumerBodyProps> = React.memo(
     if (topConsumerStatsResult.length) {
       const maxCapacityConverted = getMaxCapacity(topConsumerStatsResult);
       const sortedResult = topConsumerStatsResult.sort(sortResources);
-      const legends = topConsumerStatsResult.map((resource) => ({
-        name: getMetricType(resource, metricType),
-      }));
 
       const chartData = getGraphVectorStats(sortedResult, metricType, maxCapacityConverted.unit);
 
       const chartLineList = chartData.map((data, i) => (
         <ChartLine key={i} data={data as DataPoint[]} /> // eslint-disable-line react/no-array-index-key
       ));
+
+      // sorts the line data whose capacity are equal according to the name
+      const sortFunction = (obj) => {
+        obj.sort(function(a, b) {
+          if (a.props.data[0].y !== b.props.data[0].y) {
+            return 0;
+          }
+          const nameA = a.props.data[0].name.toLowerCase();
+          const nameB = b.props.data[0].name.toLowerCase();
+
+          if (nameA < nameB)
+            // sort string ascending
+            return -1;
+          if (nameA > nameB) return 1;
+          return 0; // default return value (no sorting)
+        });
+        return obj;
+      };
+
+      const chartLineListSorted = sortFunction(chartLineList);
```

Comment:
Cant we do sorting on the `chartData` and then build `legends` data out of it ?

Reply:
What is the reason for sorting on `chartLineListSorted` ?

Reply:
ack